spec/unit/appmgmt_spec.rb

```diff
@@ -574,6 +574,11 @@ class { cons: consume => Cap[cap]}
         }.to raise_error(/Only application components can appear inside a site - Notify\[fail me\] is not allowed at line 20/)
       end
     end
+
+    it "includes code_id if specified" do
+      catalog = compile_to_env_catalog(MANIFEST_WITH_SITE, "12345")
+      expect(catalog.code_id).to eq("12345")
+    end
   end
```

Comment:
suggest adding  tests that check that the behaviour when a new code_id is returned in the catalog is the expected, and when code_id is nil.